roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env python
+'''
+  Send Cluster EFK checks to Zagg
+'''
+# pylint: disable=missing-docstring
+# vim: expandtab:tabstop=4:shiftwidth=4
+#
+#   Copyright 2015 Red Hat Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# This is not a module, but pylint thinks it is.  This is a command.
+# pylint: disable=invalid-name
+# If a check throws an exception it is failed and should alert
+# pylint: disable=bare-except
+import ssl
+import urllib2
+import json
+
+from openshift_checks import OpenShiftCheck
+
+
+class EFKStackCheck(OpenShiftCheck):
+    """"
+        This is a check for the entire EFK stack shipped with OCP
+        https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py
+    """
+
+    name = "efk_stack_check"
+    tags = ["preflight", "elasticsearch", "efkstack"]
+
+    namespace = "logging"
+    config_file = "/etc/origin/master/admin.kubeconfig"
+
+    args = None
+    es_pods = []
+    fluentd_pods = []
+
+    def __init__(self, module_executor):
+        super(EFKStackCheck, self).__init__(module_executor)
+
+        es_cert = '/etc/elasticsearch/secret/admin-'
+        self.es_curl = "curl -s --cert {}cert --key {}key --cacert {}ca -XGET ".format(es_cert, es_cert, es_cert)
+
+    def run(self, _, task_vars):
```

Comment:
As in the other PR, I'd suggest the explicit:

```python
def run(self, tmp, task_vars):
```

I am still to see the case where `tmp` is actually used... we need to dig again into action plugins to find an example of that.

```
$ grep -nr tmp ansible/plugins/action
./async.py:32:    def run(self, tmp=None, task_vars=None):
./async.py:37:        result = super(ActionModule, self).run(tmp, task_vars)
./async.py:45:        if not tmp:
./async.py:46:            tmp = self._make_tmp_path(remote_user)
./async.py:47:            self._cleanup_remote_tmp=True
...
./unarchive.py:33:    def run(self, tmp=None, task_vars=None):
./unarchive.py:38:        result = super(ActionModule, self).run(tmp, task_vars)
./unarchive.py:62:        if not tmp:
./unarchive.py:63:            tmp = self._make_tmp_path(remote_user)
./unarchive.py:64:            self._cleanup_remote_tmp = True
...
```

There are several examples where `tmp` is used to determine where to create temp files.